cpp/daal/src/algorithms/dtrees/gbt/regression/gbt_regression_model_builder.cpp

```diff
@@ -47,6 +47,12 @@ services::Status ModelBuilder::convertModelInternal()
     return daal::algorithms::gbt::internal::ModelImpl::convertDecisionTreesToGbtTrees(modelImplRef._serializationData);
 }
 
+ModelBuilder::ModelBuilder()
+{
+    auto modelImpl = new gbt::regression::internal::ModelImpl();
+    _model.reset(modelImpl);
+}
```

Comment:
I think same from `cpp/daal/src/algorithms/dtrees/gbt/classification/gbt_classification_model_builder.cpp` should be applied here and in **all model builders** for consistency:
```cpp
ModelBuilder::ModelBuilder() : _nClasses(1), _nIterations(1), _model(new gbt::classification::internal::ModelImpl()) {}
```